netlib/tcp.py

```diff
@@ -708,6 +715,7 @@ def connect(self):
         self.connection = connection
         self.ip_address = Address(connection.getpeername())
         self._makefile()
+        return _closer(self)
```

Comment:
That's the wrong place to put the handler. We're only connected after we call .connect().